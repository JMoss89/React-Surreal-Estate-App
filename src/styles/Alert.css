.alert-success{
    background-color: lightgreen;
    border-radius: 3px;
    width: 70%;
    color:black;
    justify-content: center;
    display: block;
    margin:auto;
    padding: 0px 0px;
}

.alert-error{
    background-color: lightcoral;
    border-radius: 3px;
    width: 60%;
    color:black;
    justify-content: center;
    display: block;
    margin:auto
}
