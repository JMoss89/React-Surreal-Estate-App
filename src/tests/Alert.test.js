import React from 'react';
import { render, screen } from '@testing-library/react';
import Alert from '../components/Alert';

describe('Alert', () => {

    test('Displays an error message', () => {
        render(<Alert message='Error!' />);
        const errorMessage = screen.getByText('Error!');

        expect(errorMessage).toBeInTheDocument();
    });

    test('Desplays a success message', () => {
        render(<Alert message='Success!!!' success />);
        const successMessage = screen.getByText('Success!!!');

        expect(successMessage).toBeInTheDocument();
    });

    test("does not render an error or a success message if message props is empty", () => {
        const { asFragment } = render(<Alert message="" />);
        const alert = asFragment();
      
        expect(alert).toMatchSnapshot();
    });

    test("renders an error message if server fails", () => {
        const { asFragment } = render(<Alert message="Error!" />);
      
        expect(asFragment()).toMatchSnapshot();
    });
})
