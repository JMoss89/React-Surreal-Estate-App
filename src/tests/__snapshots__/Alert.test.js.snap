// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;

exports[`Alert renders an error message if server fails 1`] = `
<DocumentFragment>
  <div
    class="alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;
